includes/class-amp-travel-blocks.php

```diff
@@ -20,9 +20,162 @@ public function init() {
+			add_filter( 'rest_pre_echo_response', array( $this, 'filter_rest_pre_echo_response' ), 10, 3 );
+			add_action( 'pre_get_posts', array( $this, 'filter_pre_get_posts' ), 10, 1 );
 		}
 	}
 
+	/**
+	 * Filters search to search adventures by from and to date.
+	 *
+	 * @param WP_Query $query Query.
+	 */
+	public function filter_pre_get_posts( $query ) {
+
+		if ( ! is_admin() && is_search() ) {
+			$meta_query  = array();
+			$start_query = array();
+			$end_query   = array();
+
+			if ( ! empty( $_GET['start'] ) && ! empty( $_GET['end'] ) ) {
+				$start       = esc_attr( $_GET['start'] );
+				$end         = esc_attr( $_GET['end'] );
+				$start_query = array(
+					'relation' => 'OR',
+					array(
+						'relation' => 'AND',
+						array(
+							'relation' => 'OR',
+							array(
+								'key'     => 'amp_travel_end_date',
+								'value'   => $start,
+								'compare' => '>=',
+							),
+							array(
+								'key'     => 'amp_travel_end_date',
+								'value'   => '',
+								'compare' => '=',
+							),
+						),
+						array(
+							'key'     => 'amp_travel_start_date',
+							'value'   => $end,
+							'compare' => '<=',
+						),
+					),
+					array(
+						'relation' => 'AND',
+						array(
+							'key'     => 'amp_travel_start_date',
+							'value'   => '',
+							'compare' => '=',
+						),
+						array(
+							'key'     => 'amp_travel_end_date',
+							'value'   => $start,
+							'compare' => '>=',
+						),
+					),
+				);
+
+				// If we only have start date, only the end date is relevant.
+			} elseif ( ! empty( $_GET['start'] ) ) {
+				$start       = esc_attr( $_GET['start'] );
+				$start_query = array(
+					'relation' => 'OR',
+					array(
+						'key'     => 'amp_travel_end_date',
+						'value'   => '',
+						'compare' => '=',
+					),
+					array(
+						'key'     => 'amp_travel_end_date',
+						'value'   => $start,
+						'compare' => '>=',
+					),
+				);
+			} elseif ( ! empty( $_GET['end'] ) ) {
+				$end       = esc_attr( $_GET['end'] );
+				$end_query = array(
+					'relation' => 'OR',
+					array(
+						'relation' => 'AND',
+						array(
+							'relation' => 'OR',
+							array(
+								'key'     => 'amp_travel_start_date',
+								'value'   => '',
+								'compare' => '=',
+							),
+							array(
+								'key'     => 'amp_travel_end_date',
+								'value'   => $end,
+								'compare' => '<=',
+							),
+						),
+						array(
+							'key'     => 'amp_travel_start_date',
+							'value'   => $end,
+							'compare' => '<=',
+						),
+					),
+					array(
+						'relation' => 'AND',
+						array(
+							'key'     => 'amp_travel_end_date',
+							'value'   => $end,
+							'compare' => '<=',
+						),
+						array(
+							'key'     => 'amp_travel_end_date',
+							'value'   => '',
+							'compare' => '!=',
+						),
+					),
+				);
+			}
+
+			if ( ! empty( $end_query ) && ! empty( $start_query ) ) {
```

Comment:
You're right, actually looks like we can just use one `$meta_query` value instead of separating `$end_query` and `$start_query` and leave the `if ( ! empty( $end_query ) && ! empty( $start_query ) ) { ... ` part completely out.